blacksheep/testing/client.py

```diff
@@ -0,0 +1,96 @@
+from typing import Dict, Optional
+
+from blacksheep.contents import Content
+from blacksheep.server.application import Application
+from blacksheep.server.responses import Response
+from blacksheep.testing.simulator import AbstractTestSimulator, TestSimulator
+
+
+class TestClient:
+    # Setting this dunder variable
+    # We tell to pytest don't discover this up
+    __test__ = False
+
+    def __init__(
+        self, app: Application, test_simulator: Optional[AbstractTestSimulator] = None
+    ):
+        self._test_simulator = test_simulator or TestSimulator(app)
+
+    async def get(
```

Comment:
Looks great! Can you please add also methods for HTTP Methods: `OPTIONS`, `HEAD`, `TRACE`? I think leaving out `CONNECT` is fine, but especially `OPTIONS` and `HEAD` are relevant to test CORS and caching.

Reply:
Will do